### PR TITLE
Add indication for remaining time when validating in workbench

### LINKED ISSUE #2034: Show estimate of time remaining for upload
> Progress bars for upload progress in workbench would be another nice addition.

**Requested By:** Rob at CSIRO

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixed #2034

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx [hunk 1]
```diff
@@ -134,6 +134,44 @@ export function WbStatus({
...
+        seconds < 10 ? '0' : ''
+      }${seconds}`;
+    }
+    return '';
+  }
+
+  // Calculate the remaining time
+  React.useEffect(() => {
+    if (status.taskstatus === 'PROGRESS') {
+      if (current > 15) {
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx:155]: Don't just put magic numbers in the code.
Create a variable right above the component, and add a comment on top of the variable describing what it's used for

File: specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx [hunk 2]
```diff
@@ -134,6 +134,44 @@ export function WbStatus({
...
+        const interval = setInterval(() => {
+          const remainingSeconds = Math.round(
+            (((total - current) / current) * (Date.now() - startTime.current)) /
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx:158]: (optional) this expression could be split into multiple variables to make it more readable

File: specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx [hunk 3]
```diff
@@ -134,6 +134,44 @@ export function WbStatus({
       </>
     );
 
+  const startTime = React.useRef(Date.now());
+  const [remainingTime, setRemainingTime] = React.useState<number | null>(null);
+
+  // Convert seconds to minutes and seconds and create returned string
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx:140]: This function should not be defined inside a component as it doesn't have any logic specific to this component - it's just a utility. Move it into `./Atoms/Internalization.tsx` or `./utils/utils.ts`

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx (hunk 1 above)
Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/components/WorkBench/Status.tsx:155]: For context, wiki page about [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))

**Review by maxxxxxdlp — APPROVED** [review]

**Review by chanulee1 — APPROVED** [review]
Timer for validation present.

**Comment by specifysoftware** [comment]
This pull request has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/specify-7-8-7-release-announcement/1079/1

**Pull request merged**